### ISSUE TITLE
Can this code run on multiple GPUs?

### ISSUE THREAD

**Issue by lxc86739795** [CompVis/metric-learning-divide-and-conquer#4]
It seems that the code cannot directly run on multiple GPUs by simply adding "torch.nn.Dataparallel".

**Comment by asanakoy** [comment]
I'm not sure about that. Maybe you will have to make some extra modifications.
What kind of error do you get?

**Comment by Chen94yue** [comment]
LOL @lxc86739795

**Comment by faizwhb** [comment]
The simplest way to go about this is to make modifications until train_batch and disable faiss_gpu. You might have to make more modification in terms of gpu sharing between faiss and the training session.
